app/controllers/application_controller.rb

```diff
@@ -321,8 +321,10 @@ def saved_redirect_url_or(default)
   end
 
   def generic_exception(exception)
-    Foreman::Logging.exception("Action failed", exception)
-    render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception => exception}
+    exception = Foreman::WrappedException.new(exception, exception.message) unless exception.is_a? Foreman::Exception
+    ex_message = exception.message
+    Foreman::Logging.exception(ex_message, exception)
+    render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception_code => exception.code, :exception_message => ex_message, request_id: request.request_id.split('-').first}
```

Comment:
Oh I see, you want to utilize exception code? This will not work like that. These codes are precaldulated from our codebase, using that will actually generate you the same error code over and over again for the same input message but with a different callstacks.

What you want to do is to simply generate some kind of UUID or any kind of random token that is unique enough and log that into log. What I think is much better is show a request UUID from logging context, which is unique and safe to use (it is not the Rails request UUID which is different).